lib/src/combined_wrappers.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// Returns a new list that represents [lists] flattened into a single list.
+///
+/// All methods and accessors treat the new list as-if it were a single
+/// concatenated list, but the underlying implementation is based on lazily
+/// accessing individual list instances.
+///
+/// The returned list is unmodifiable.
+List/*<T>*/ combineLists/*<T>*/(List<List/*<T>*/> lists) {
```

Comment:
> Isn't the fact that it is a `CombinedListView` an implementation detail?

I think a class better expresses what's actually happening. Functions generally produce outputs that are, after the function is finished running, largely independent of their arguments. Classes are much more likely to retain a reference to their constructor argument and update their state based on changes, and the word "View" explicitly indicates the way that will happen.

It's also more consistent with similar existing patterns—we have `Unmodifiable*View` classes, not `make*Unmodifiable()` functions.

> I could have it return an `UnmodifiableListView<T>` though.

*Hopefully* no one is relying on that static type anywhere.